src/app/notes/notes.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit } from '@angular/core';
+import { NoteService } from '../services/note.service';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Note } from '../models/note';
+
+@Component({
+  selector: 'app-notes',
+  templateUrl: './notes.component.html',
+  styleUrls: ['./notes.component.scss']
+})
+export class NotesComponent implements OnInit {
+  displayedColumns: string[] = ['id', 'name', 'note', 'category', 'delete'];
+  
+  noteForm = new FormGroup({
+    name: new FormControl('', Validators.required),
+    note: new FormControl('', Validators.required),
+    category: new FormControl('', Validators.required)
+  });
+  
+  constructor(public noteService: NoteService) {}
+
+  ngOnInit() {
+    this.noteService.currentNote && this.noteForm.patchValue(this.noteService.currentNote); 
+  }
+
+  setCurrentNote(note: Note) {
+    this.noteService.setCurrentNote(note);
+    this.noteService.currentNote && this.noteForm.patchValue(this.noteService.currentNote); 
+  }
+
+  addNote() {
+    this.noteService.addNote();
+    this.noteService.currentNote && this.noteForm.patchValue(this.noteService.currentNote); 
+  }
+
+  saveNote() {
+    if (this.noteForm.invalid) {
```

Comment:
Do we need this check? This should never be true because the event should never fire unless it's valid